server.js

```diff
@@ -1,32 +1,152 @@
-import express from 'express'
-import bodyParser from 'body-parser'
+import express, { response } from 'express'
 import cors from 'cors'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+import topMusicData from './data/top-music.json'
+
+const ERROR_MESSAGE_SONG = {error:'No song fund, try an other title.'}
+
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors())
-app.use(bodyParser.json())
+app.use(express.json());
 
-// Start defining your routes here
+// dummy endpoint
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.json('🖤 Music endpoint starts here! 🖤, List of all traks with info: /music. List of all music genres: /music/allgenres. All songs by one artist or a song by title:music/info?title=(songs title) or music/info?artist=U(name of artist) Get all songs from an Artist no matter spelling: /music/artists/:artist(add name). Get all songs from a genre /music/genre/pop (or rock or something else). All songs with same energy: /music/energy/ and a number. Sorted The 20 least popular songs, low to high: /music/popularity/lowest-20. Sorted The 20 most popular songs, hight to low; /music/popularity/best-20. ' )
 })
 
+//all songs endpoint http://localhost:8080/music
+app.get('/music', (req, res) => {
+  const { song } = req.query
+  if (song) {
+    const musicList = topMusicData.filter(song => song.trackName.includes(song))
+    res.json(musicList)
+  } 
+  res.json(topMusicData)
+})
+
+
+// app.get('/music/info', (req, res) => {
+
+//   const { song, artist } = req.query
+
+//   if (song) {
+//     const songByTitle = topMusicData.filter(song => song.trackName.includes(song))
+//     if (songByTitle.length === 0){
+//       res.status(404).json(ERROR_MESSAGE_SONG)
+//     } else {
+//       res.json(songByTitle)
+//     }
+//   } else if (artist){
+//     const artistName = topMusicData.filter((item) => {
+//     return item.artistName.toLowerCase().indexOf(artist.toLowerCase()) !== -1;
+// })
+//     res.json({ data: artistName });
+//   } 
+  
+// })
+
```

Comment:
what happened here? 🕵🏻

Reply:
What do you mean? It's an endpoint that sort either by song title or artist name. Or do you mean that it is commented out?